ansible_navigator/configuration/application_configuration.py

```diff
@@ -0,0 +1,199 @@
+""" the ansible-navigator configuration
+"""
+import os
+
+from .application_post_processor import ApplicationPostProcessor
+from .definitions import Config
+from .definitions import CliParameters
+from .definitions import SubCommand
+from .definitions import Entry
+from .definitions import EntryValue
+
+from ..utils import oxfordcomma
+
+
+def abs_user_path(fpath):
+    """Resolve a path"""
+    return os.path.abspath(os.path.expanduser(fpath))
+
+
+def generate_editor_command():
+    """Generate a default for editor_command if EDITOR is set"""
+    if "EDITOR" in os.environ:
+        command = "%s {filename}" % os.environ.get("EDITOR")
+    else:
+        command = "vi +{line_number} {filename}"
+    return command
+
+
+PLUGIN_TYPES = (
+    "become",
+    "cache",
+    "callback",
+    "cliconf",
+    "connection",
+    "httpapi",
+    "inventory",
+    "lookup",
+    "module",
+    "netconf",
+    "shell",
+    "strategy",
+    "vars",
+)
+
+ApplicationConfiguration = Config(
+    application_name="ansible-navigator",
+    post_processor=ApplicationPostProcessor(),
+    subcommands=[
+        SubCommand(name="collections", description="Explore available collections"),
+        SubCommand(name="config", description="Explore the current ansible configuration"),
+        SubCommand(name="doc", description="Review documentation for a module or plugin"),
+        SubCommand(name="inventory", description="Explore an inventory"),
+        SubCommand(name="load", description="Explore a playbook artifact"),
+        SubCommand(name="run", description="Run a playbook"),
+    ],
+    entries=[
+        Entry(
+            name="app",
+            description="Placeholder for subcommand name",
+            subcommand_value=True,
+            value=EntryValue(default="welcome"),
+        ),
+        Entry(
+            name="cmdline",
+            description="Placeholder for argparse remainder",
+            value=EntryValue(default=[]),
+        ),
+        Entry(
+            name="container_engine",
+            choices=["podman", "docker"],
+            cli_parameters=CliParameters(short="--ce"),
+            description="Specify the container engine",
+            value=EntryValue(default="podman"),
```

Comment:
The default value and the description strings can be read from a separate config file `(anisble-navigator.yaml or ansible-navigator.json)` which can be stored under `ansible_navigator/configuration` file.

Reply:
It can help kee the docs and config option within code in sync

Reply:
I think I prefer it all in one place, this file/object can be loaded by python and the entries iterated to generate a doc

Reply:
I think it might also leave the door open for problems in the future, right now an "Entry" cannot be defined without a description since it's required by the underlying obj definition, if it gets pulled if could be forgotten.....  So we would have to author some tests to check for that.....  the key's would have to stay in sync as well since each entry would need a corresponding entry in the external file, I'm not thinking it's worth it here.